crates/corro-agent/src/agent/util.rs

```diff
@@ -1223,3 +1225,92 @@ pub fn check_buffered_meta_to_clear(
 
     conn.prepare_cached("SELECT EXISTS(SELECT 1 FROM __corro_seq_bookkeeping WHERE site_id = ? AND version >= ? AND version <= ?)")?.query_row(params![actor_id, versions.start(), versions.end()], |row| row.get(0))
 }
+
+pub async fn clear_empty_versions(
+    agent: Agent,
+    actor_id: ActorId,
+    feedback: Option<&Sender<String>>,
+) -> eyre::Result<()> {
+    if let Some(ref tx) = feedback {
+        tx.send(format!("compacting empty versions for {actor_id}"))
+            .await?
+    }
+
+    let start = Instant::now();
+
+    let query = r#"SELECT start_version FROM __corro_bookkeeping WHERE actor_id = :actor_id AND end_version IS null AND db_version
+    NOT IN (SELECT DISTINCT db_version FROM crsql_changes WHERE site_id = :actor_id)"#;
+
+    let conn = agent.pool().read().await?;
+
+    let overwritten_versions: RangeInclusiveSet<Version> = conn
+        .prepare(query)?
+        .query_map(named_params! {":actor_id": actor_id}, |row| {
+            Ok(row.get(0)?..=row.get(0)?)
+        })?
+        .collect::<rusqlite::Result<RangeInclusiveSet<Version>>>()?;
+
+    let total = overwritten_versions.len();
+
+    if let Some(ref tx) = feedback {
+        tx.send(format!(
+            "got {total} ranges to clear for actor {actor_id} in {:?}",
+            start.elapsed()
+        ))
+        .await?
+    }
+    debug!("got {total} ranges to clear for actor {actor_id}");
+
+    // process empty version in chunks of 10
+    // for chunks in overwritten_versions.iter().chunks(10) {
+    //     let mut conn = agent.pool().write_low().await?;
+    //     let tx = conn.immediate_transaction()?;
+    //     for v in ch {
+    //         store_empty_changeset(&tx, actor_id, v)?;
+    //     }
+    //     tx.commit()?;
+    //
+    //     // yield a bit at the end
+    //     tokio::task::yield_now().await;
+    // }
+
+    let mut chunk: Vec<RangeInclusive<Version>> = vec![];
+    for v in overwritten_versions {
+        chunk.push(v);
+
+        if chunk.len() == 10 {
+            let mut conn = agent.pool().write_low().await?;
+            let tx = conn.immediate_transaction()?;
+            for v in chunk {
+                store_empty_changeset(&tx, actor_id, v)?;
+            }
+            tx.commit()?;
+            chunk = vec![];
+            // tokio::task::yield_now().await;
+        }
+    }
+
+    // the rest in the chunks
+    {
+        let mut conn = agent.pool().write_low().await?;
+        let tx = conn.immediate_transaction()?;
+        for v in chunk {
+            store_empty_changeset(&tx, actor_id, v)?;
+        }
+        tx.commit()?;
+    }
```

Comment:
```suggestion
    for chunk in overwritten_versions.chunks(10) {
        let mut conn = agent.pool().write_low().await?;
        let tx = conn.immediate_transaction()?;
        for v in chunk {
            store_empty_changeset(&tx, actor_id, v)?;
        }
        tx.commit()?;
    }
```

Reply:
`Iterator::chunks` comes from the itertools crate. https://docs.rs/itertools/latest/itertools/trait.Itertools.html#method.chunks